### PR TITLE
Added getDates flow with redux.

### PR DESCRIPTION
**Pull request by ddomb**

### CODE REVIEW

File: app/frontend/src/components/Calendar/components/Cells.jsx [hunk 1]
```diff
@@ -74,13 +80,23 @@ const dateLabelStatus = {
...
+    const dateMonthStart = startOfMonth(currentDate);
+    const dateMonthEnd = endOfMonth(dateMonthStart);
+    const dateStartDate = startOfWeek(dateMonthStart);
+    const dateEndDate = endOfWeek(addWeeks(dateMonthEnd, 1));
+    return {monthStart: dateMonthStart,
+            monthEnd: dateMonthEnd,
+            startDate: dateStartDate,
+            endDate: dateEndDate};
+  })
+  const fetchParams = {start: getUnixTime(monthStart),
```
Comment (IamShobe) [inline app/frontend/src/components/Calendar/components/Cells.jsx]: Move it inline or wrap with use memo..

File: app/frontend/src/services/date_datas.js [hunk 2]
```diff
@@ -24,6 +24,30 @@ class DateStatusService {
...
+          }
+        }
+      );
+      return response.data;
+    
+    } catch (error) {
+      if (error.response.status === 401) {
+        throw new PermissionsError(error.response.data.details);
+      }
+      console.warn("couldn't get reasons", error.response.status);
```
Comment (IamShobe) [inline app/frontend/src/services/date_datas.js]: Reasons?

File: app/frontend/src/screens/Dashboard/Dashboard.jsx [hunk 3]
```diff
@@ -56,6 +56,7 @@ const Dashboard = React.memo((props) => {
       changeSelectedValue(null);
     }
   });
+  const userIdList = [useSelector(state => lodash.get(state.users.me, "id"))];
```
Comment (IamShobe) [inline app/frontend/src/screens/Dashboard/Dashboard.jsx]: Wrap it in list in the inner most u can - get params of fetch.
Change it to user ID
The calendar never showing details of multiple users.

Comment (ddomb) [reply app/frontend/src/screens/Dashboard/Dashboard.jsx]: I made it from here so the calendar will be generic, we want to use the calendar in different locations (commander space for example) that we don't necessarily want to get the user data but we want to get other users.

Comment (IamShobe) [reply app/frontend/src/screens/Dashboard/Dashboard.jsx]: yes i agree its here.
but instead of a list do it a simple var.
there is no need for a list

Comment (IamShobe) [reply app/frontend/src/screens/Dashboard/Dashboard.jsx]: also it is strange that the calendar needs more than 1 man if we always display one

Comment (ddomb) [reply app/frontend/src/screens/Dashboard/Dashboard.jsx]: If I remember correctly I need to send list if users id in the get request

Comment (ddomb) [reply app/frontend/src/screens/Dashboard/Dashboard.jsx]: In the past we talked about showing all the soldiers calendar in the commander space (all of them in one calendar

Comment (IamShobe) [reply app/frontend/src/screens/Dashboard/Dashboard.jsx]: really?

Comment (IamShobe) [reply app/frontend/src/screens/Dashboard/Dashboard.jsx]: ok then..

**Pull request merged**